fastparse/shared/src/test/scala/fastparse/ExampleTests.scala

```diff
@@ -158,6 +158,16 @@ object ExampleTests extends TestSuite{
           failure.trace == """xml:0 / rightTag:7 / "abcde":9 ..."edcba>""""
         )
       }
+      'filter{
+        val digits = P(CharIn('0' to '9').rep(1).!).map(_.toInt)
+        val even = digits.filter(_ % 2 == 0)
+        val Result.Success(12, _) = even.parse("12")
+        val failure = even.parse("123").asInstanceOf[Result.Failure]
```

Comment:
Could we add asserts for `Failure#trace` as well as `even.toString` to make sure they're reasonable? I'd expect `even.toString` to equal `digits.filter(<function1>)` in this case since we're passing in an anonymous lambda